### PR TITLE
[parametric] Add Java library OTel support

### PR DESCRIPTION
**Pull request by PerfectSlayer**
## Description

This PR adds OpenTelemetry support to Java library client for parametric tests.
It requires some `dd-trace-java` instrumentation fixes (https://github.com/DataDog/dd-trace-java/pull/5037).

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Sone lint is required : `./format.sh`

File: parametric/test_otel_span_methods.py
[the remark this answers is not in the document]
Comment (PerfectSlayer) [reply parametric/test_otel_span_methods.py:1]: This is expected as it requires a new release from the Java tracer.

> It requires some dd-trace-java instrumentation fixes (https://github.com/DataDog/dd-trace-java/pull/5037).

I plan to merge this PR as #1021 only once tracer is released.

[the remark this answers is not in the document]
Comment (cbeauchesne) [reply parametric/test_otel_span_methods.py:1]: ok, I'll set it as a draft then.  You can set it back to nornal once you need a review

[the remark this answers is not in the document]
Comment (PerfectSlayer) [reply parametric/test_otel_span_methods.py:1]: Ok. The PR is complete though.
I will mark it as ready once we did a release then.

**Comment by PerfectSlayer** [comment]
The `1.12.0` version of Java tracer was just released. 
So it is ready for review @cbeauchesne

**Comment by cbeauchesne** [comment]
relaunching the CI !

**Comment by PerfectSlayer** [comment]
> It's still failing on java/parametric, is it expected ?

Looks like nobody is maintaining the docker image with the tracer... 
I will need to fix it myself or revert to the curl/wget solution to download the Java tracer.

But in the same time, the OTel tests changed again... I will deal with it later I guess 🤷

**Comment by PerfectSlayer** [comment]
@cbeauchesne I updated it with the old way to get tracer and latest parametric tests state.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**